src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/AssemblyPart.cs

```diff
@@ -41,15 +41,37 @@ public AssemblyPart(Assembly assembly)
         public IEnumerable<TypeInfo> Types => Assembly.DefinedTypes;
 
         /// <inheritdoc />
-        public IReadOnlyList<CompilationLibrary> GetCompilationLibraries()
+        public IEnumerable<string> GetReferencePaths()
         {
             var dependencyContext = DependencyContext.Load(Assembly);
             if (dependencyContext != null)
             {
-                return dependencyContext.CompileLibraries;
+                var referencePaths = new List<string>();
+                var compileLibraries = dependencyContext.CompileLibraries;
+
+                try
+                {
+                    for (var i = 0; i < compileLibraries.Count; i++)
+                    {
+                        var compileLibrary = compileLibraries[i];
+                        referencePaths.AddRange(compileLibrary.ResolveReferencePaths());
+                    }
+
+                    return referencePaths;
+                }
+                catch
```

Comment:
>  I don't think there's much a user would be to able to do with the exception.

But doesn't this indicate a deployment/publishing problem? We could possibly provide a better error